plugin.rb

```diff
@@ -344,82 +345,72 @@ def self.skip_db?
-      posts.where(
-        "topics.id IN (
-        SELECT tc.topic_id
-        FROM topic_custom_fields tc
-        WHERE tc.name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}' AND
-                        tc.value IS NOT NULL
-        )",
+  solved_callback = ->(scope) do
+    sql = <<~SQL
+      topics.id IN (
+        SELECT topic_id
+          FROM topic_custom_fields
+         WHERE name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}' 
+           AND value IS NOT NULL
       )
-    end
+    SQL
 
-    Search.advanced_filter(/status:unsolved/) do |posts|
-      if SiteSetting.allow_solved_on_all_topics
-        posts.where(
-          "topics.id NOT IN (
-          SELECT tc.topic_id
-          FROM topic_custom_fields tc
-          WHERE tc.name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}' AND
-                          tc.value IS NOT NULL
-          )",
-        )
-      else
-        tag_ids = Tag.where(name: SiteSetting.enable_solved_tags.split("|")).pluck(:id)
-
-        posts.where(
-          "topics.id NOT IN (
-          SELECT tc.topic_id
-          FROM topic_custom_fields tc
-          WHERE tc.name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}' AND
-                          tc.value IS NOT NULL
-          ) AND (topics.id IN (
-            SELECT top.id
-            FROM topics top
-            INNER JOIN category_custom_fields cc
-            ON top.category_id = cc.category_id
-            WHERE cc.name = '#{::DiscourseSolved::ENABLE_ACCEPTED_ANSWERS_CUSTOM_FIELD}' AND
-                          cc.value = 'true'
-          ) OR topics.id IN (
-            SELECT top.id
-            FROM topics top
-            INNER JOIN topic_tags tt
-            ON top.id = tt.topic_id
-            WHERE tt.tag_id IN (?)
-          ))",
-          tag_ids,
+    scope.where(sql)
+  end
+
+  unsolved_callback = ->(scope) do
+    scope = scope.where <<~SQL
+      topics.id NOT IN (
+        SELECT topic_id
+          FROM topic_custom_fields
+         WHERE name = '#{::DiscourseSolved::ACCEPTED_ANSWER_POST_ID_CUSTOM_FIELD}' 
+           AND value IS NOT NULL
+      )
+    SQL
+
+    if !SiteSetting.allow_solved_on_all_topics
+      tag_ids = Tag.where(name: SiteSetting.enable_solved_tags.split("|")).pluck(:id)
+
+      scope = scope.where <<~SQL, tag_ids
+        topics.id IN (
+          SELECT t.id
+            FROM topics t
+            JOIN category_custom_fields cc 
+              ON t.category_id = cc.category_id
+             AND cc.name = '#{::DiscourseSolved::ENABLE_ACCEPTED_ANSWERS_CUSTOM_FIELD}' 
+             AND cc.value = 'true'
+        ) 
+        OR 
+        topics.id IN (
+          SELECT topic_id 
+            FROM topic_tags 
+           WHERE tag_id IN (?)
         )
-      end
+      SQL
     end
+
+    scope
   end
 
-  if Discourse.has_needed_version?(Discourse::VERSION::STRING, "1.8.0.beta6")
+  if respond_to? :register_custom_filter_by_status
```

Comment:
Using the new API introduced in https://github.com/discourse/discourse/pull/26770